{% extends 'base.html' %}

{% block title %}Gerenciar Visitantes{% endblock %}

{% block content %}
<div class="container mt-4">
    <h2 class="text-center">Gerenciar Visitantes</h2>

    <div class="text-center mb-4">
        <a href="{% url 'cadastrar_ou_editar_visitante' %}" class="btn btn-primary">Cadastrar Novo Visitante</a>
    </div>

    {% if visitantes.exists %}
    <div class="mt-4">
        <h3>Visitantes Cadastrados</h3>
        <ul class="list-group">
            {% for visitante in visitantes %}
            <li class="list-group-item d-flex justify-content-between align-items-center">
                <span>{{ visitante.nome }} - {{ visitante.instituicao }} - {{ visitante.email }}</span>
                <div>
                    <a href="{% url 'cadastrar_ou_editar_visitante' visitante.id %}" class="btn btn-sm btn-secondary">Editar</a>
                    <button type="button" class="btn btn-sm btn-danger" onclick="confirmarExclusao('{{ visitante.nome }}', '{{ visitante.instituicao }}', {{ visitante.id }})">Excluir</button>
                </div>
            </li>
            {% endfor %}
        </ul>
    </div>
    {% else %}
    <p class="text-center">Nenhum visitante cadastrado.</p>
    {% endif %}
</div>

<script>
    function confirmarExclusao(nome, instituicao, visitanteId) {
        if (confirm(`Deseja prosseguir com a exclusão do visitante ${nome} da ${instituicao}?`)) {
            // Faz uma solicitação AJAX para excluir o visitante
            fetch(`/usuarios/excluir_visitante/${visitanteId}/`)
                .then(response => response.json())
                .then(data => {
                    if (data.status === 'error') {
                        // Exibe um pop-up de erro
                        alert(data.message);
                    } else if (data.status === 'success') {
                        // Exibe um pop-up de sucesso e recarrega a página
                        alert(data.message);
                        window.location.reload();
                    }
                })
                .catch(error => {
                    console.error("Erro ao excluir visitante:", error);
                    alert("Ocorreu um erro inesperado. Tente novamente mais tarde.");
                });
        }
    }
</script>
{% endblock %}
